### DISCUSSION TITLE
Deploying to a Server Tips?

### DISCUSSION THREAD

**Discussion by collinskatie** [empiricaly/empirica#527] [General]
Dear Empirica Community, 

We are trying to deploy our experiment to a server (using AWS EC2) and had two questions. 

1) When we run empirica serve {our tar file}, we can open the empirica page thru port 3000 of the public IP instance. However, we can't seem to access the admin panel? It says "No Experiments Available" and then when we open "3000/admin" are encountered with a username/password log-in (but can't seem to log-in?) Is there a way to create/reset an account, or anything else we should be doing to add in the batches while running on the server? 

2) If we make changes to the experiment while running on the server, do we need to construct a new bundle + re-scp it to the server? Are there good practices for dynamically adjusting the experiment while in the server-phase (or are any edits best done just thru localhost experimentation). 

Thanks for any tips!

**Comment by npaton** [comment]
To answer your questions (sorry for the delay):

1. When developing locally, we disable username/password for the admin. In production, it's required. A username ("admin") and a random password are generated when you create your project. You can find (and change them) them in the `.empirica/empirica.toml` file in your project.
2. If you make changes, you will need to re-bundle, upload your bundle again, and restart the server. If you want to tweak configuration, the preferred method is to make your experiment configurable using factors/treatments.

I would also recommend you host your experiment behind a reverse proxy as described here: https://docs.empirica.ly/guides/deploying-my-experiment#optional-setup-a-reverse-proxy

That page contains other useful information about deployment.

Cheers!
